Dockerfiles/feast-operator.Dockerfile.konflux

```diff
@@ -0,0 +1,30 @@
+# Build the manager binary
+FROM registry.access.redhat.com/ubi8/go-toolset:1.22.9 AS builder
+ARG TARGETOS
+ARG TARGETARCH
+
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# Copy the go source
+COPY cmd/main.go cmd/main.go
+COPY api/ api/
+COPY internal/controller/ internal/controller/
+
+# Build
+# the GOARCH has not a default value to allow the binary be built according to the host where the command
+# was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
+# the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
+# by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
+RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o manager cmd/main.go
+
+FROM registry.access.redhat.com/ubi8/ubi-micro:8.10
+WORKDIR /
+COPY --from=builder /opt/app-root/src/manager .
+USER 65532:65532
+
+ENTRYPOINT ["/manager"]
```

Comment:
why copy the Dockerfile to another location? this will make maintenance difficult going forward and will cause confusion in the future for sure

Reply:
surely you could just set the working/build dir to `infra/feast-operator`

Reply:
is the idea to protect the downstream builds from upstream changes? if so, how will this change impact our future stable branch syncs? is this limited to a release branch?

Reply:
@tchughesiv We create our own copy of Dockerfile so we are able to independently do konflux related changes to it. This is a standard practice for all RHOAI components